proto/babylonlabs/babylon/v1beta1/tx.proto

```diff
@@ -13,11 +13,61 @@ option (gogoproto.goproto_getters_all) = false;
 service Msg {
   option (cosmos.msg.v1.service) = true;
 
+  // InstantiateBabylonContracts defines an operation for instantiating the
+  // Babylon contracts.
+  rpc InstantiateBabylonContracts(MsgInstantiateBabylonContracts)
+      returns (MsgInstantiateBabylonContractsResponse);
+
   // UpdateParams defines a (governance) operation for updating the x/auth
   // module parameters. The authority defaults to the x/gov module account.
   rpc UpdateParams(MsgUpdateParams) returns (MsgUpdateParamsResponse);
 }
 
+// MsgInstantiateBabylonContracts is the Msg/InstantiateBabylonContracts request
+// type.
+message MsgInstantiateBabylonContracts {
+  option (cosmos.msg.v1.signer) = "signer";
+
+  // signer is the address who submits the message.
+  string signer = 1 [ (cosmos_proto.scalar) = "cosmos.AddressString" ];
+
+  // babylon_contract_code_id is the code ID for the Babylon contract.
+  uint64 babylon_contract_code_id = 2;
+  // btc_staking_contract_code_id is the code ID for the BTC staking contract.
+  uint64 btc_staking_contract_code_id = 3;
+  // btc_finality_contract_code_id is the code ID for the BTC finality contract.
+  uint64 btc_finality_contract_code_id = 4;
```

Comment:
I would just do all this internally. Why storing the contracts externally, and then calling this to do the instantiation, when everything could be done internally during bootstrapping?
This is the mesh-security approach, which I think makes sense as the SDK is useless without the contracts. The contracts can in fact be thought of as part of the SDK.

Reply:
Hmm from what I've seen in mesh security sdk, 

- Contracts are written as parameters https://github.com/osmosis-labs/mesh-security-sdk/blob/b507993c9799de4a513b8e75ea2659e9bf659779/x/meshsecurityprovider/types/genesis.pb.go#L28-L32
- The keeper initialisation function does not store/instantiate the contracts https://github.com/osmosis-labs/mesh-security-sdk/blob/b507993c9799de4a513b8e75ea2659e9bf659779/x/meshsecurityprovider/keeper/keeper.go#L31-L45
- in e2e tests the contracts are deployed in a separate step https://github.com/osmosis-labs/mesh-security-sdk/blob/b507993c9799de4a513b8e75ea2659e9bf659779/tests/e2e/test_client.go#L405-L436

So it seems that mesh security also relies on someone to instantiate contracts in advance, and then pass a gov prop to let the parameters point to these addresses?

Reply:
Oh, you're right. I was confused with tgrade, should have taken a look first, sorry. So, I think that yes, your proposal makes sense. Will review the rest next.

Reply:
I was also confused by this, which I saw once

https://github.com/osmosis-labs/mesh-security-sdk/blob/b507993c9799de4a513b8e75ea2659e9bf659779/tests/starship/setup/main.go#L140

But, it belongs to a test setup. The approach you follow is valid as well.